src/api/app/models/event_subscription.rb

```diff
@@ -13,59 +25,22 @@ def subscriber
     end
   end
 
-  def receiver_role
-    read_attribute(:receiver_role).to_sym
-  end
-
-  def self._get_role_rule(relation, role)
-    all_rule = nil
-    relation.each do |r|
-      if r.receiver_role == role
-        return r
-      end
-      if r.receiver_role == :all
-        all_rule = r
-      end
+  def subscriber=(subscriber)
```

Comment:
I know this can be polymorphic but I prefer to do that in a seperate PR after this one because there are already quite a lot of changes in this PR..